### ISSUE TITLE
Start MQTT connection after WiFi is connected

### ISSUE THREAD

**Issue by andriensis** [HeimdallMidgard/ESP-32-BLE-Scanner#6]
Hi! I'm testing the library on a few of my ESP32 at home.

I have noticed the MQTT connection is started when the device has booted but that usually leads to MQTT errors because the device has not connected to the WiFi yet. I believe it should wait for a successful WiFi connection and only then start the MQTT connection.

I could send a pull request (already doing this on my device) if you agree. 
Let me know!

**Comment by HeimdallMidgard** [comment]
Hey, yeah if you tested it feel free to send a pull request and I will test it and build a new binary afterwards. There are still some problems regarding the Wifi connection I will take a look at in the future. Thanks for sharing it. It's always nice to code together.

**Comment by andriensis** [comment]
What kind of issues did you notice with the wifi connection?
I've noticed some variables change value after a while. Like the wifi name or password lose its json value and change to something unreadable which therefore doesn't work when attemping to connect to the wifi.

**Comment by HeimdallMidgard** [comment]
Yeah I saw the same issue. But I saw this issue only on newer ESP32 platform. Older version doesn't seem to have this issue but I wanted to run it on the newer version.
